pkg/app/pipectl/prompt/reader.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prompt
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+const (
+	maximum_retry = 5 // maximum number of retry for required fields
+)
+
+// Reader reads input from user or mock.
+type Reader struct {
+	bufReader bufio.Reader
+}
+
+func NewReader(in io.Reader) Reader {
+	return Reader{
+		bufReader: *bufio.NewReader(in),
+	}
+}
+
+// ReadString reads a string with showing message. If the input is empty, it returns an empty string.
+// If the input is more than one word, it returns an error.
+func (r *Reader) ReadString(message string) (string, error) {
+	s, e := r.ReadStrings(message)
+	if e != nil {
+		return "", e
+	}
+
+	if len(s) == 0 {
+		return "", nil
+	}
+	if len(s) > 1 {
+		return "", fmt.Errorf("too many arguments")
+	}
+
+	return s[0], nil
+}
```

Comment:
[imo] How about rename like below? WDYT? 
- `Reader` -> `Prompt`
- `ReadXX` -> `InputXX`

Reply:
Thanks, I renamed 
 - `Reader`  -> `Prompt`
 - `ReadXX` -> removed and created `prompt.Input` struct.